portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,27 +75,59 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
   /** Returns comments fetched from datastore */
   private ArrayList<Comment> fetchComments(HttpServletRequest request) {
     //Prepare datastore query
-    Query query = new Query("Comment");
-    PreparedQuery results = datastore.prepare(query);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     
     //Set limit options
     FetchOptions fetchOptions = FetchOptions.Builder.withDefaults();
     Optional<Integer> limit = getLimit(request);
     if (limit.isPresent()) {
       fetchOptions = FetchOptions.Builder.withLimit(limit.get());
     }
+    
+    //Set username filter
+    Optional<String> username = getUsername(request);
+    if (username.isPresent()) {
+      query.addFilter("user", FilterOperator.EQUAL, username.get());
+    }
 
     //Create comment list
+    PreparedQuery results = datastore.prepare(query);
     ArrayList<Comment> comments = new ArrayList<>();
     for (Entity entity: results.asIterable(fetchOptions)) {
-      String content = (String) entity.getProperty("content");
-      String id = (String) entity.getProperty("id");
-      Comment comment = new Comment(id, content);
+      Comment comment = getCommentFromEntity(entity);
       comments.add(comment);
     }
     return comments;
   }
 
+  /** Returns a Comment object constructed from the given entity. */
+  private Comment getCommentFromEntity(Entity entity) {
```

Comment:
"get" usually implies retrieving the data from some other source. A better name might be `transformCommentToEntity`.

Same for the method below that does the opposite.